ansible/roles/gitlab_runner/tasks/main.yml

```diff
@@ -78,32 +78,36 @@
   register: gitlab_runner__register_packages
   until: gitlab_runner__register_packages is succeeded
 
-- name: Check if GitLab service is available
-  ansible.builtin.uri:
-    url: '{{ gitlab_runner__api_url }}'
-  register: gitlab_runner__register_api
-  when: gitlab_runner__api_url | d() and gitlab_runner__api_url
-  failed_when: False
-  no_log: '{{ debops__no_log | d(True) }}'
-
 - name: Register new GitLab Runners
   ansible.builtin.uri:
-    url: '{{ (item.api_url | d(gitlab_runner__api_url)) + "api/v4/runners" }}'
+    url: '{{ (item.api_url | d(gitlab_runner__api_url)) + "api/v4/user/runners" }}'
     method: 'POST'
-    body: 'token={{ item.token | d(gitlab_runner__token) }}&description={{ item.name
-           | urlencode }}&tag_list={{ ((item.tags | d([])
-                                        + (gitlab_runner__shell_tags
-                                           if (item.executor == "shell") else [])
-                                        + gitlab_runner__combined_tags) | unique | join(","))
-           | urlencode }}&run_untagged={{ item.run_untagged | d(gitlab_runner__run_untagged) }}'
+    headers:
+      'PRIVATE-TOKEN': '{{ item.api_token | d(gitlab_runner__api_token) }}'
+    body_format: 'form-urlencoded'
+    body:
+      runner_type: '{{ item.runner_type | d(gitlab_runner__runner_type) }}'
+      group_id: '{{ item.group_id | d(gitlab_runner__group_id) | d(omit) }}'
+      project_id: '{{ item.project_id | d(gitlab_runner__project_id) | d(omit) }}'
+      description: '{{ item.name }}'
+      tag_list: '{{ (item.tags | d([])
+                     + (gitlab_runner__shell_tags
+                        if (item.executor == "shell") else [])
+                     + gitlab_runner__combined_tags)
+                     | unique | join(",") }}'
+      run_untagged: '{{ item.run_untagged | d(gitlab_runner__run_untagged) }}'
+      paused: '{{ item.paused | d(omit) }}'
+      locked: '{{ item.locked | d(omit) }}'
+      access_level: '{{ item.access_level | d(omit) }}'
+      maximum_timeout: '{{ item.maximum_timeout | d(omit) }}'
+      maintenance_note: '{{ item.maintenance_note | d(omit) }}'
```

Comment:
Are we sure that the `ansible.builtin.url` Ansible module does a correct thing here and removes the parameters with the `omit` value from the request, or is the current value of `omit` passed along the request? If it is, are there any issues in GitLab when this happens?

Reply:
I'm using this version to manage the gitlab-runner 17.0+ and it works without issues.

Quick check, I've modify [uri.py](https://github.com/ansible/ansible/blob/v2.16.7/lib/ansible/modules/uri.py) to this:
```patch
+   module.fail_json(msg="Encoded body: %s" % to_native(body))
+
    # Make the request
    start = utcnow()
    r, info = uri(module, url, dest, body, body_format, method,
                  dict_headers, socket_timeout, ca_path, unredirected_headers,
                  decompress, ciphers, use_netrc)
```
Result:
```
msg: 'Encoded body: runner_type=group_type&group_id=887490&description=gl-runner5&tag_list=managed-by-debops&run_untagged=True'
```